src/DotVVM.Framework/Configuration/DotvvmConfiguration.cs

```diff
@@ -269,6 +270,18 @@ public static DotvvmConfiguration CreateDefault(IServiceProvider serviceProvider
             return config;
         }
 
+        /// <summary>
+        /// Creates a configuration with fake services in place of hosting-specific components.
+        /// </summary>
+        internal static DotvvmConfiguration CreateInternal()
+        {
+            return CreateDefault(services =>
+            {
+                services.TryAddSingleton<IViewModelProtector, FakeViewModelProtector>();
+                services.TryAddSingleton<ICsrfProtector, FakeCsrfProtector>();
+            });
+        }
+
```

Comment:
We have methods like this all over the place. I think it's time we added it into Framework itself.

Reply:
It sounds legit.